frontend/src/hooks/useLastViewedFlags.ts

```diff
@@ -0,0 +1,55 @@
+import { useCallback, useEffect, useState } from 'react';
+import { getLocalStorageItem, setLocalStorageItem } from '../utils/storage';
+import { basePath } from 'utils/formatPath';
+import { useCustomEvent } from './useCustomEvent';
+
+const MAX_ITEMS = 3;
+
+type Flag = { featureId: string; projectId: string };
+
+const removeIncorrect = (flags?: any[]): Flag[] => {
+    if (!Array.isArray(flags)) return [];
+    return flags.filter((flag) => flag.featureId && flag.projectId);
+};
+
+const localStorageItems = (key: string) => {
+    return removeIncorrect(getLocalStorageItem(key) || []);
+};
+
+export const useLastViewedFlags = () => {
```

Comment:
stores last 3 flags in local storage and propagates the changes to other usage of this hook